content-security-policy/reporting/report-uri-from-child-frame.html

```diff
@@ -8,17 +8,13 @@
 </head>
 <body>
     <script nonce="abc">
-      var t1 = async_test("Check that we received a message from the child frame");
```

Comment:
I believe this was here precisely to have a declared test so the testing harness won't immediately conclude the test when the page loads. I think this test now will succeed immediately without waiting for a message.

Reply:
In the absence of any tests, testharness.js will not react to the `load` event. Instead, it will wait for the global `done` function to be invoked (as an indication of a single-page test) or for a sub-test to be defined. This test defines one sub-test via `checkReport.sub.js`, and since that is the only test it defines, there's no need to explicitly manage scheduling.

More than being superfluous, the removed sub-test was a source of instability. It completed before the meaningful sub-test was defined, so there was a race condition regarding when testharness.js determined completion.

Reply:
Ah ok, I did not know it behaves like that when there are no tests.